src/main/java/com/uber/cadence/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -197,7 +207,45 @@ public Result handle(
         metricsScope.gauge(MetricsType.ACTIVITY_ACTIVE_THREAD_COUNT).update(Thread.activeCount());
       }
     }
-    return activity.execute(activityTask, metricsScope);
+
+    ActivityTaskHandler.Result activityResult;
```

Comment:
I added a test for activity timeouts, but for local activities I am facing some issues since actually the local activities seem to not have timeouts specified and instead there is some mechanism on timing out a the whole decision.